content/ghc-9.12-cabal-3.14/index.md

```diff
@@ -0,0 +1,128 @@
++++
+title = "GHC 9.12 & cabal 3.14 releases"
+date = 2024-12-22
+[taxonomies]
+authors = ["Hécate"]
+categories = ["Cabal", "GHC"]
+tags = ["Release"]
++++
+
+To conclude the year 2024, the GHC and Cabal teams are happy to announce the releases of GHC 9.12 and cabal 3.14.
+
+<!-- more -->
+
+
+Here are some highlights:
+
+## GHC 9.12
+
+### Language Extensions
+
+This release brings many new and exciting extensions:
+* [Multiline string literals][MultilineStrings] to write string literals with newlines kept verbatim,
+  enabling multiline strings without the need for string gaps and literal \n characters";
+* [Or-Patterns][OrPatterns], allowing you to match on several patterns in a single `case` branch:
+
+```haskell
+data Sweet = Cupcake | Liquorice | Cookie | Raisins
+
+-- Without Or-Patterns
+tasty Cupcake = True
+tasty Cookie = True
+tasty _ = False
+
+-- With Or-Patterns
+tasty (Cupcake; Cookie) = True
+tasty (Liquorice; Raisins) = False
+```
+
+* [NamedDefaults][NamedDefaults] allows you to create type defaulting declarations for literals other than for the `Num` class. This mechanism is used to make sure that in the expression `print (6 + 7)`, the multiplication gets a concrete type like `Integer`, `Int` or `Double`, instead of the vague  `Num a => a`.  
+You can now apply this mechanism to other typeclasses, like:
+
+```haskell
+default IsString (Text, String)
+```
+
+Which means that in a module with [`OverloadedStrings`][OverloadedStrings], string literals `"like this"` will default to [`Text`][Text]
+instead of triggering an error about a polymorphic type.
+
+### Base Library
+
+* Improvements to backtraces and exceptions. Lots of them.
+
+#### ⚠️ Deprecations
+
+* The deprecation period for [GHC.Pack][GHC.Pack] has come to an end. Goodbye!
+* [GHC.Desugar][GHC.Desugar] will be removed in GHC 9.14.
```

Comment:
What does this actually mean that 'a deprecation period' has ended? It's not crystal clear whether it means the deprecation was cancelled, the module was removed in 9.14, the module was previously removed before 9.14 or whether it will now get removed, after 9.14 was released. The valediction 'Goodbye' is cute but unnecessary, it would be better to just plainly state that the module is not available in 9.14.

I think it would be better to avoid colloquial or memey language like e.g. the '[friendship has ended, now best friend](https://knowyourmeme.com/memes/friendship-ended-with-mudasir)' meme. It's culture specific and unnecessary, hasn't traditionally been used (at least not according to [releases](https://github.com/haskell/cabal/releases)) and I don't understand why now is the time to start using memes.